app/modules/notifications/models.py

```diff
@@ -189,6 +197,31 @@ def update_preference(self, notification_type, notification_channel, status):
         assert isinstance(status, bool)
         self.preferences[notification_type][notification_channel] = status
 
+    @classmethod
+    def validate_preferences(cls, new_prefs):
```

Comment:
We had a discussion in discord on Friday about using marshmallow to validate json and I tried it out in #286.  There's a lot of code for validation :neutral_face: but the declaration of the schema means it's a bit easier to see how the json should look like.